lib/hotel.rb

```diff
@@ -0,0 +1,120 @@
+module Hotel
+  class Hotel
+    attr_reader :number_of_rooms, :list_of_rooms, :reservations
+
+    def initialize(number_of_rooms)
+      @number_of_rooms = number_of_rooms.to_i
+      @reservations = []
+      @list_of_rooms = Array.new
+      @number_of_rooms.times do |i|
+        @list_of_rooms << Room.new(i+1)
+      end
+    end
+
+    def make_reservation(client, arrival_date, departure_date, number_of_rooms)
+      available_rooms = available_at_period(arrival_date, departure_date,)
+      if available_rooms == nil
+        raise NoAvailableRoomError.new("No avaibility at these dates.")
+      elsif available_rooms.length < number_of_rooms.to_i
+        raise NoAvailableRoomError.new("Only #{available_rooms.length} rooms available at these dates.")
+      else
+        number_of_rooms.to_i.times do |i|
+          available_at_period(arrival_date, departure_date,).first.booked << {"arrival" => Date.parse(arrival_date.sub(/[\/]/, '-')), "departure" => Date.parse(departure_date.sub(/[\/]/, '-'))}
+        end
+      end
+      @reservations << Reservation.new(client, arrival_date, departure_date, number_of_rooms)
+    end
+
+    def create_block(client, arrival_date, departure_date, number_of_rooms)
+      available_rooms = available_at_period(arrival_date, departure_date)
+      if find_block(client) != nil
+        raise ArgumentError.new("This client already has a room block")
+      elsif available_rooms == nil
+        raise NoAvailableRoomError.new("No avaibility at these dates.")
+      elsif available_rooms.length < number_of_rooms.to_i
+        raise NoAvailableRoomError.new("Only #{available_rooms.length} rooms available at these dates.")
+      else
+        number_of_rooms.to_i.times do |i|
+          available_at_period(arrival_date, departure_date).first.booked << {"arrival" => Date.parse(arrival_date.sub(/[\/]/, '-')), "departure" => Date.parse(departure_date.sub(/[\/]/, '-'))}
```

Comment:
this line is very long!

Also, instead of shoveling a very specific hash into a property of a `room`, it might be better to move this logic into the room itself. We want our room to know how to add the information to the `booked` property, not hotel